### PR TITLE
F# RFC FS-1020 - byref returns

### PR DESCRIPTION
**Pull request by dsyme**
See https://github.com/fsharp/FSharpLangDesign/blob/master/RFCs/FS-1020-byref-returns.md

### CODE REVIEW

**Comment by dsyme** [comment]
This feature is now code-complete and the testing added (subject to getting CI to pass)

File: src/fsharp/CompileOps.fs [hunk 1]
```diff
@@ -4976,10 +4976,13 @@ module private ScriptPreprocessClosure =
     
         // Mark the last file as isLastCompiland. 
         let closureFiles =
-            match List.frontAndBack closureFiles with
-            | rest, ClosureFile(filename,m,Some(ParsedInput.ImplFile(ParsedImplFileInput(name,isScript,qualNameOfFile,scopedPragmas,hashDirectives,implFileFlags,_))),errs,warns,nowarns) -> 
-                rest @ [ClosureFile(filename,m,Some(ParsedInput.ImplFile(ParsedImplFileInput(name,isScript,qualNameOfFile,scopedPragmas,hashDirectives,implFileFlags,(true, tcConfig.target.IsExe)))),errs,warns,nowarns)]
-            | _ -> closureFiles
+            if isNil closureFiles  then
```
Comment (dsyme) [inline src/fsharp/CompileOps.fs]: This is a fix unrelated to the feature - I noticed that when calling "fsi foo.fsx" when foo.fsx doesn't exist we hit a crash in ``List.frontAndBack``.

**Comment by KevinRansom** [comment]
@dsyme I see a couple of merge issues can you take care of it pls.

Thanks

**Comment by KevinRansom** [comment]
@dsyme.Thank you for this contribution

**Pull request merged**